src/game/client/gui/gamewindowtransitions.h

```diff
@@ -0,0 +1,62 @@
+/**
+ * @file
+ *
+ * @author tomsons26
+ *
+ * @brief Classes for handling WND UI system transitions.
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#pragma once
+#include "always.h"
+#include "namekeygenerator.h"
+#include "subsysteminterface.h"
+#include <list>
+
+class GameWindow;
+class Transition;
+
+enum WindowTransitionStyle
+{
+    TRANSITION_FLASH,
+    TRANSITION_BUTTON_FLASH,
+    TRANSITION_FADE,
+    TRANSITION_SCALE_UP,
+    TRANSITION_MAIN_MENU_SCALE_UP,
+    TRANSITION_TYPE_TEXT,
+    TRANSITION_SCREEN_FADE,
+    TRANSITION_COUNT_UP,
+    TRANSITION_FULL_FADE,
+    TRANSITION_TEXT_ON_FRAME,
+    TRANSITION_MAIN_MENU_MEDIUM_SCALE_UP,
+    TRANSITION_MAIN_MENU_SMALL_SCALE_DOWN,
+    TRANSITION_CONTROL_BAR_ARROW,
+    TRANSITION_SCORE_SCALE_UP,
+    TRANSITION_REVERSE_SOUND,
+};
+
+class TransitionGroup;
```

Comment:
You can move this up to the other class declarations.

Reply:
the actual class will replace that later